File: ClientAPI/src/main/java/pt/upskill/clientapi/CustomExceptions/UserNotFoundException.java
package pt.upskill.clientapi.CustomExceptions;

/**
 * Exception thrown when a user is not found in the system.
 * This exception extends {@link RuntimeException} and is used to
 * indicate that an expected user could not be located.
 */
public class UserNotFoundException extends RuntimeException {

    /**
     * Default constructor for UserNotFoundException.
     * Calls the superclass's default constructor.
     */
    public UserNotFoundException() {
        super();
    }

    /**
     * Constructor for UserNotFoundException with a custom message.
     *
     * @param message the detailed message to be passed to the exception
     */
    public UserNotFoundException(String message) {
        super(message);
    }
}

File: ForumAPI/ForumAPI/CustomExceptions/UserNotFoundException.cs
namespace ForumAPI.CustomExceptions
{
    /// <summary>
    /// Custom exception that is thrown when a requested user is not found in the system.
    /// </summary>
    public class UserNotFoundException : Exception
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="UserNotFoundException"/> class with a specified error message.
        /// </summary>
        /// <param name="message">The message that describes the error.</param>
        public UserNotFoundException(string message) : base(message) { }

        /// <summary>
        /// Initializes a new instance of the <see cref="UserNotFoundException"/> class with a default error message.
        /// </summary>
        public UserNotFoundException() : base()
        {
            // Default constructor that can be used when no specific message is provided.
        }
    }
}